packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -74,12 +75,14 @@ export class ConnectionService {
    * @param config config for creation of connection request
    * @returns outbound message containing connection request
    */
-  public async protocolCreateRequest(
+  public async createRequest(
     outOfBandRecord: OutOfBandRecord,
     config: ConnectionRequestParams
   ): Promise<ConnectionProtocolMsgReturnType<ConnectionRequestMessage>> {
-    // TODO check oob role is sender
-    // TODO check oob state is await-response
+    this.logger.debug(`Create message ${ConnectionRequestMessage.type} start`, outOfBandRecord)
+    outOfBandRecord.assertRole(OutOfBandRole.Receiver)
+    outOfBandRecord.assertState([OutOfBandState.Initial, OutOfBandState.PrepareResponse])
```

Comment:
In what case is it initial and what case is it prepareResponse?

Reply:
It didn't make sense. I updated it.

Reply:
I wasn't using initial and setting it directly to one of prepare/await response. So the check should be more strict. But I updated oob module to set oob state to `initial` in `createInvitation` and then update it in `acceptInvitation`. The method `recieveInvitation` stays same, setting `await-response` directly to a new oob record.